src/io/terminal.cpp

```diff
@@ -76,3 +76,30 @@ void terminal_write(const char* data) {
 		terminal_putchar(data[i]);
 }
 
+void terminal_write_int(int n) {
+	if (n < 0) {
+		terminal_putchar('-');
+		n = n * -1;
+	}
+
+	// Can shorten this step when we implement log
+	int num_digits = 0;
+	int tmp = n;
+	do {
+		num_digits++;
+		tmp = tmp / 10;
+	} while (tmp > 0);
+
+	do {
+		// Can shorten this step when we implement pow
+		int base = 1;
+		for (int i = 0; i < num_digits - 1; i++) {
+			base = base * 10;
```

Comment:
ok, but for real, I think you mixed base 10 and base generic code. also, you should be able to use integer division